figures/5.figure5/scripts/2.add_scale_bars.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import tifffile as tf  # write tiff files
+from PIL import Image  # read tiff files
+from toml import load
+from tqdm import tqdm  # progress bar
+
+# In[2]:
+
+
+CELL_TYPE = "PBMC"
+
+
+# In[3]:
+
+
+path = pathlib.Path(
+    "../../../8.cytopick_analysis/results/PBMC/single_cell_predictions.parquet"
+).resolve()
+
+# read in the data
+df = pd.read_parquet(path)
+# show all columns
+pd.set_option("display.max_columns", None)
+df.head()
+
+
+# In[5]:
+
+
+for image_path in df["image_crop_path"]:
+    print(image_path)
+    # open the image
+    image = cv2.imread(image_path, cv2.IMREAD_UNCHANGED)
+    # image to array
+    image_array = np.array(image)
+    resolution = 0.1550  # pixels per micrometer
+    # get the image size
+    image_size = image_array.shape
+    if image_array.shape[1] <= 100:
+        scale_bar_length = 5  # um
```

Comment:
How did you identify these? Just for future reference and implementation, how would you find these values?

Reply:
this is the crop size that I set. So for larger cells if the crops are set to say 250 pixel crops then change the value. :)